src/flash/display/Loader.ts

```diff
@@ -83,77 +82,98 @@ module Shumway.AVM2.AS.flash.display {
     };
     static initializer: any = function() {
       var self: Loader = this;
-      DisplayObject._advancableInstances.push(self);
+      display.DisplayObject._advancableInstances.push(self);
     };
 
     static classSymbols: string [] = null;
     static instanceSymbols: string [] = null;
 
-
     /**
-     * Handles the load status and dispatches progress events. This gets manually triggered in the
-     * event loop to ensure the correct order of operations.
+     * In each turn of the event loop, Loader events are processed in two batches:
+     * first INIT and COMPLETE events are dispatched for all active Loaders, then
+     * OPEN and PROGRESS.
+     *
+     * A slightly weird result of this is that INIT and COMPLETE are dispatched one
+     * turn *later* than the other events: INIT is only dispatched if the loadStatus
```

Comment:
> This is not necessarily related.

This only seems relevant if `COMPLETE` is dispatched in the same turn as the last `PROGRESS` event for loads that take longer than one turn of the event loop (i.e. where `INIT` and `COMPLETE` are dispatched in different turns.) I didn't test that and it doesn't seem too relevant, so I'd argue for keeping the current order of dispatching events, where we don't need to do any additional checks with newly introduced flags.

> It also looks like that processing loading events for the root loader takes one complete turn exclusively, meaning all loaders created on the very first frame are getting processed starting from frame 2. That really only is the case on the very first turn of the event loop. All loaders created in frames after that, are processed within the same turn.

If you have a test case for this, please send it my way. I don't fully understand the description in the abstract.